src/Microsoft.VisualStudio.Sdk.TestFramework.Xunit/BrokeredServiceContractTestBase`3.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Diagnostics;
+using System.IO.Pipelines;
+using Microsoft.ServiceHub.Framework;
+using Microsoft.VisualStudio.Threading;
+using Nerdbank.Streams;
+using Xunit;
+
+namespace Microsoft.VisualStudio.Sdk.TestFramework;
+
+/// <summary>
+/// A base class for testing Visual Studio brokered service contracts with client callbacks.
+/// </summary>
+/// <typeparam name="TInterface">The service interface.</typeparam>
+/// <typeparam name="TServiceMock">The class that mocks the service.</typeparam>
+/// <typeparam name="TClientInterfaceMock">The class that mockes the service's client callback.</typeparam>
+public abstract class BrokeredServiceContractTestBase<TInterface, TServiceMock, TClientInterfaceMock> : BrokeredServiceContractTestBase<TInterface, TServiceMock>
+    where TInterface : class
+    where TServiceMock : TInterface, IMockServiceWithClientCallback, new()
+    where TClientInterfaceMock : class, new()
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="BrokeredServiceContractTestBase{TInterface, TServiceMock, TClientInterfaceMock}"/> class.
+    /// </summary>
+    /// <param name="logger"><inheritdoc cref="LoggingTestBase(ITestOutputHelper)" path="/param[@name='logger']"/></param>
+    /// <param name="serviceRpcDescriptor">The descriptor that the product will use to request or proffer the brokered service.</param>
+    public BrokeredServiceContractTestBase(ITestOutputHelper logger, ServiceRpcDescriptor serviceRpcDescriptor)
+        : base(logger, serviceRpcDescriptor)
+    {
+        Assert.NotNull(serviceRpcDescriptor.ClientInterface);
```

Comment:
This isn't the test itself. It's a test framework. Use `Requires` for argument validation rather than xunit's `Assert` class.